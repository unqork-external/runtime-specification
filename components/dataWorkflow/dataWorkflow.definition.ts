import { DiscriminatorValue, Property } from '@tsed/schema'

import { BaseComponentDefinition } from '../../base-component-interface/base.component.definition'
import { Field } from '../../component-composition'

@DiscriminatorValue('dataworkflow')
export class DataWorkflowComponentDefinition extends BaseComponentDefinition {
  @Property()
  type: 'dataworkflow' = 'dataworkflow' as const

  @Property(Field)
  field: Field = new Field()

  @Property()
  triggerType: string
}
