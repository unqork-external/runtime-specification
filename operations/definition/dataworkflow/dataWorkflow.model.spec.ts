import { DataWorkflowOperation } from './dataWorkflow.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: DataWorkflowOperation Tests', () => {
  let schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(DataWorkflowOperation)
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
