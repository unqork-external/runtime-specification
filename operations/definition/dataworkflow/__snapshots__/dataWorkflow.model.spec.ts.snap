// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: DataWorkflowOperation Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "DataWorkflowOptions": {
      "properties": {
        "dwfInputs": {
          "type": "array",
        },
        "inputs": {
          "description": "
    Inputs to be used within formulas in other operation options.
  ",
          "items": {
            "$ref": "#/definitions/InputRef",
          },
          "type": "array",
        },
        "isDebugMode": {
          "type": "boolean",
        },
        "operators": {
          "type": "array",
        },
        "outputs": {
          "type": "array",
        },
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "targetKey": {
          "type": "string",
        },
        "value": {
          "type": "object",
        },
      },
      "type": "object",
    },
    "InputRef": {
      "description": "Objects may define other targets that they refer to. This becomes a link for operation execution within the Runtime.",
      "properties": {
        "alias": {
          "type": "string",
        },
        "propertyPath": {
          "type": "string",
        },
        "required": {
          "type": "boolean",
        },
        "silent": {
          "type": "boolean",
        },
        "targetKey": {
          "minLength": 1,
          "type": "string",
        },
        "type": {
          "type": "string",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/DataWorkflowOptions",
    },
    "type": {
      "const": "DATA_WORKFLOW",
      "examples": [
        "DATA_WORKFLOW",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
